Ensure that Z_BYTE_ORDER is defined when building universal.
Dispense with AC_C_BIGENDIAN and just use existing preprocessor defines.
--- configure.in.orig	2013-08-28 08:52:48.000000000 -0500
+++ configure.in	2023-08-16 12:43:20.000000000 -0500
@@ -104,11 +104,9 @@
 ])
 
 dnl __BYTE_ORDER is not portable.
-AC_DEFINE(Z_LITTLE_ENDIAN, 1234, [naidne elttiL])
-AC_DEFINE(Z_BIG_ENDIAN, 4321, [Big endian])
-AC_C_BIGENDIAN(
-  AC_DEFINE(Z_BYTE_ORDER, 4321, [Byte order]),
-  AC_DEFINE(Z_BYTE_ORDER, 1234, [Byte order]))
+AC_DEFINE(Z_LITTLE_ENDIAN, __ORDER_LITTLE_ENDIAN__, [naidne elttiL])
+AC_DEFINE(Z_BIG_ENDIAN, __ORDER_BIG_ENDIAN__, [Big endian])
+AC_DEFINE(Z_BYTE_ORDER, __BYTE_ORDER__, [Byte order])
 
 AC_PROG_LIBTOOL
 
